src/jit/lir.cpp

```diff
@@ -1110,6 +1113,28 @@ bool LIR::Range::CheckLIR(Compiler* compiler, bool checkUnusedValues) const
     // Make sure the range itself is valid.
     assert(IsValid());
 
+    // Make sure the list is well-formed: no node appears more than once (no circularities). Insert each node in a set
```

Comment:
Yes. I think if you call Insert*(SeqTree()) on a tree that points to existing LIR nodes this can happen.